crates/ruff_python_parser/src/lexer/ftstring.rs

```diff
@@ -4,31 +4,40 @@ use super::TokenFlags;
 
 /// The context representing the current f-string that the lexer is in.
 #[derive(Clone, Debug)]
-pub(crate) struct FStringContext {
+pub(crate) struct FTStringContext {
+    kind: FTStringKind,
     flags: TokenFlags,
 
-    /// The level of nesting for the lexer when it entered the current f-string.
+    /// The level of nesting for the lexer when it entered the current f/t-string.
     /// The nesting level includes all kinds of parentheses i.e., round, square,
     /// and curly.
     nesting: u32,
 
-    /// The current depth of format spec for the current f-string. This is because
+    /// The current depth of format spec for the current f/t-string. This is because
     /// there can be multiple format specs nested for the same f-string.
     /// For example, `{a:{b:{c}}}` has 3 format specs.
     format_spec_depth: u32,
 }
 
-impl FStringContext {
-    pub(crate) const fn new(flags: TokenFlags, nesting: u32) -> Self {
-        assert!(flags.is_f_string());
+impl FTStringContext {
+    pub(crate) const fn new(kind: FTStringKind, flags: TokenFlags, nesting: u32) -> Self {
+        match kind {
+            FTStringKind::FString => assert!(flags.is_f_string()),
+            FTStringKind::TString => assert!(flags.is_t_string()),
+        }
```

Comment:
Unrelated to your changes but I think I'd prefer to change the constructor to

```rust
new(flags: TokenFlags, nesting: u32) -> Option<Self>
```

https://github.com/astral-sh/ruff/blob/fa628018b28fa018d7a0e109223186172ef91275/crates/ruff_python_parser/src/lexer.rs#L634-L636

can then call `new` and only call into `lex_fstring` if the return value is `Some`